webapp/src/extensionsBrowser.tsx

```diff
@@ -408,71 +405,148 @@ export const ExtensionsBrowser = (props: ExtensionsProps) => {
 
     const categoryNames = getCategoryNames();
     const local = currentTab == TabState.InDevelopment ? fetchLocalRepositories() : undefined
-
-    return <div className={`extensionsBrowser ${props.isVisible ? "" : "hide"}`} >
-        {showImportExtensionDialog ? <ImportModal onCancelClick={() => setShowImportExtensionDialog(false)} onImportClick={handleImportUrl} /> : undefined}
-        {deletionCandidate ? <DeleteConfirmationModal ns={deletionCandidate.name} onCancelClick={() => { setDeletionCandidate(undefined) }} onDeleteClick={() => { removeDepAsync(deletionCandidate)}} /> : undefined}
-        <MenuBar className="extensionsHeader" ariaLabel={lf("Extentions")}>
-            <div className="header-left">
-                <Button className="menu-button" leftIcon="fas fa-arrow-left large" title={lf("Back")} label={lf("Back")} onClick={props.hideExtensions} />
-            </div>
-            <div className="header-center">
-                {lf("Extensions")}
-            </div>
-            <div className="header-right"></div>
-        </MenuBar>
-        <div className="extensionSearchHeader">
-            <div className="header">{(lf("Do more with your micro:bit"))}</div>
-            <SearchInput searchHandler={setSearchFor} />
-            <div className="extensionTags">
-                {categoryNames.map(c =>
-                    <div className={"extensionTag " + (selectedTag == c ? "selected" : "")} onClick={() => handleCategoryClick(c)}>{c}</div>
-                )}
-            </div>
-            {/* TODO bring in the import modal in later! <div className="importButton">
-                <span>{lf("or ")}</span>
-                <div className="importButtonLink" onClick={() => setShowImportExtensionDialog(true)}>{lf("import extension")}</div>
-            </div> */}
-        </div>
-        {displayMode == ExtensionView.Search &&
-            <div className="extension-display">
-                <div className="breadcrumbs">
-                    <span className="link" onClick={handleHomeButtonClick}>{lf("Home")}</span>
-                </div>
-                <div className="extension-grid">
-                    {extensionsToShow?.map(scr =>
-                        <ExtensionCard scr={scr} onCardClick={installExtension} learnMoreUrl={scr.fullName ? `/pkg/${scr.fullName}` : undefined}
-                            name={scr.name} imageUrl={scr.imageUrl} description={scr.description} loading={scr.loading} />)}
-                </div>
-            </div>}
-        {displayMode == ExtensionView.Tags &&
-            <div className="extension-display">
-                <div className="breadcrumbs">
-                    <span className="link" onClick={handleHomeButtonClick}>{lf("Home")}</span>
-                    <span>/</span>
-                    <span>{selectedTag}</span>
+    let loadingCardId = 0
```

Comment:
i don't think you need this variable. in the callbacks to map below, an index is also passed in after the element